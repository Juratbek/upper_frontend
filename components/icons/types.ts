export interface IIconProps {
  color?: string;
  variant?: 'outlined' | 'fulfilled';
  width?: number;
  height?: number;
  opacity?: number;
}
