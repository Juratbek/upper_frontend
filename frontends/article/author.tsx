import { SidebarBlog } from 'components';
import { FC } from 'react';
import { IBlogMedium } from 'types';

import { IAuthorProps } from './article.types';

const blog: IBlogMedium = {
  id: 1,
  name: 'Blog name',
  imgUrl: '',
  bio: 'Blog bio will be here',
  articlesCount: 10,
  followersCount: 23,
};

export const Author: FC<IAuthorProps> = (props) => {
  return <SidebarBlog {...props} {...blog} />;
};
