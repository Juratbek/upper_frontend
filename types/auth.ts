import { Profile } from 'next-auth';
import { BuiltInProviderType } from 'next-auth/providers';
import { ClientSafeProvider, LiteralUnion } from 'next-auth/react';
import { AUTH_PROVIDER_TYPES } from 'variables';

type TGitHubUserType = 'User';

export type TAuthProviderProp = Profile & Record<string, unknown>;

export type TAuthProviderTypes =
  | typeof AUTH_PROVIDER_TYPES.google
  | typeof AUTH_PROVIDER_TYPES.facebook
  | typeof AUTH_PROVIDER_TYPES.github;

export type TAuthProviders = Record<LiteralUnion<BuiltInProviderType, string>, ClientSafeProvider>;

export interface IAuthProvider {
  [name: string]: (profile: TAuthProviderProp) => boolean;
}

export interface IGitHubUser {
  login: string;
  id: number;
  node_id: string;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: TGitHubUserType;
  site_admin: boolean;
  name?: string;
  company?: string;
  blog: string;
  location?: string;
  email?: string;
  hireable?: boolean;
  bio?: string;
  twitter_username?: string;
  public_repos: number;
  public_gists: number;
  followers: number;
  following: number;
  created_at: string;
  updated_at: string;
}

export interface IGoogleUser {
  iss: string;
  azp: string;
  aud: string;
  sub: string;
  email: string;
  email_verified: boolean;
  at_hash: string;
  name: string;
  picture: string;
  given_name: string;
  family_name: string;
  locale: string;
  iat: number;
  exp: number;
}

export interface IFacebookUser {
  id: string;
  name: string;
  email: string;
  picture: {
    data: {
      height: number;
      is_silhouette: boolean;
      url: string;
      width: number;
    };
  };
}

export interface ITelegramUser {
  auth_date: number;
  first_name: string;
  last_name?: string;
  photo_url: string;
  hash: string;
  id: number;
  username: string;
}
