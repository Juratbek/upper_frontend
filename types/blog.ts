import { ILink } from './common';

export interface IBlogSmall {
  id: number;
  imgUrl: string;
  name: string;
}
export interface IBlogMedium extends IBlogSmall {
  bio?: string;
  followersCount?: number;
  articlesCount?: number;
}

export interface IBlog extends IBlogMedium {
  links: ILink[];
  createdDate: string;
  isFollowed?: boolean;
}
