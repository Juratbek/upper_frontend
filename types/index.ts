export * from './common';
export * from './article';
export * from './blog';
export * from './label';
export * from './notification';
export * from './auth';
