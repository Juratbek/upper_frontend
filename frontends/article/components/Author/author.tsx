import { Avatar, Link } from 'components/lib';
import { SubscriptionButton, UnsubscribeModal } from 'components/molecules';
import { FC } from 'react';

import classes from './Author.module.scss';
import { IAuthorProps } from './author.types';

export const Author: FC<IAuthorProps> = ({ name, bio, imgUrl, id }) => {
  return (
    <div className={classes.root}>
      <Link href={`/blogs/${id}`} className={classes.author}>
        <Avatar size='medium' imgUrl={imgUrl} />
        <div className={classes['text-block']}>
          <h3 className={classes.name}>{name} sahna stend up</h3>
          <p className={classes.bio}>{bio} Lorem ipsum dolor sit.</p>
        </div>
      </Link>
      <SubscriptionButton className={classes.subscribe} blogId={id} />
      <UnsubscribeModal />
    </div>
  );
};
