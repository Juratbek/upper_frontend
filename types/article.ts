import { OutputData } from '@editorjs/editorjs';

import { IBlog } from './blog';
import { ILabel } from './label';

export interface IArticleResult {
  id: number;
  title: string;
  content: string;
  imgUrl?: string;
  author?: IBlog;
  labels: ILabel[];
  publishedDate: Date;
  updatedDate?: Date;
  viewCount?: number;
  likes?: number;
}

export interface IArticle {
  id: number;
  title: string;
  content: OutputData;
  imgUrl?: string;
  author?: IBlog;
  labels: ILabel[];
  publishedDate: string;
  updatedDate?: string;
  viewCount?: number;
  likes?: number;
  prevArticleId?: number;
  nextArticleId?: number;
}
