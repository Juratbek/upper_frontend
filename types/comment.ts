import { IBlogSmall } from './blog';

export interface IComment {
  id: number;
  date: string;
  text: string;
  author: IBlogSmall;
}
