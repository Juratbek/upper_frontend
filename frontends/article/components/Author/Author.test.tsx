import { render } from '@testing-library/react';
import { QueryWrapper, ReduxWrapper } from 'tests';

import { Author } from './author';
import { IAuthorProps } from './author.types';

const mock: IAuthorProps = {
  id: 12,
  imgUrl: 'https://upper.uz/image',
  name: "Jur'atbek",
  bio: 'React developer',
};

describe('Author component', () => {
  it('snapshot', () => {
    render(
      <ReduxWrapper>
        <QueryWrapper>
          <Author {...mock} />
        </QueryWrapper>
      </ReduxWrapper>,
    );
    expect(document.body).toMatchSnapshot();
  });
});
