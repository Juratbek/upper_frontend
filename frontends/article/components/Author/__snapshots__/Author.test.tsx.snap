// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Author component > snapshot 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <a
        class="author link"
        href="/web/blogs/12"
      >
        <div
          class="avatar avatar--medium"
          style="border-color: rgba(245, 245, 245, 0.1);"
        >
          <img
            alt="UPPER"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="http://localhost:3000/_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fupper.uz%2Fimage&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="text-block"
        >
          <h3
            class="name"
          >
            Jur'atbek
          </h3>
          <p
            class="bio"
          >
            React developer
          </p>
        </div>
      </a>
      <div
        class="modal"
        data-testid="modal-root"
      >
        <button
          class="root modal-bg"
        />
        <div
          class="modal-dialog modal-dialog--undefined"
        >
          <div
            class="modal-body"
          >
            <p
              class="text"
            >
              Obunani bekor qilmoqchimisiz?
            </p>
          </div>
          <div
            class="d-flex"
          >
            <button
              class="root unsubscribe-btn"
            >
              Bekor qilish
            </button>
            <button
              class="root no-btn"
            >
              yo'q
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
